gk/fib.c

```diff
@@ -872,13 +939,35 @@ ether_cache_put(struct gk_fib *neigh_fib,
 	}
 
 	if (addr->proto == ETHER_TYPE_IPv4) {
-		return neigh_del_ether_cache(
-			&neighbor_fib->u.neigh, &addr->ip.v4.s_addr);
+		ret = put_arp((struct in_addr *)
+			&addr->ip.v4, gk_conf->lcores[0]);
+		if (ret < 0)
+			return ret;
+
+		ret = rte_hash_del_key(neighbor_fib->u.neigh.hash_table,
+			&addr->ip.v4.s_addr);
+		if (ret < 0) {
+			RTE_LOG(CRIT, GATEKEEPER,
+				"gk: failed to delete an Ethernet cache entry with IP type %hu at %s, we are not trying to recover from this failure!",
```

Comment:
Please mention in the message that the lookup was done over the IPv4 neighbor table; otherwise, this message won't be distinguishable from the one for IPv6. Do the same change to the IPv6 message below.